dir.c

```diff
@@ -129,6 +131,25 @@ bool sqfs_dir_next(sqfs *fs, sqfs_dir *dir, sqfs_dir_entry *entry,
 	
 	*err = SQFS_OK;
 	entry->offset = dir->offset;
+
+	if (entry->offset < 2) {
+		/* offsets 0 and 1 are '.' and '..' which are not stored */
+		entry->type = SQUASHFS_DIR_TYPE;
+		if (entry->name != NULL) {
```

Comment:
Minor nit: This probably looks a lot nicer using `strncpy()` (or similar) than doing character-by-character modification. Similarly below in traverse.c, using `strcmp()`.